pkg/network/lxcusernic/lxcusernic.go

```diff
@@ -0,0 +1,156 @@
+package lxcusernic
+
+import (
+	"net"
+	"os"
+	"os/exec"
+	"strconv"
+	"time"
+
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv4/client4"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	"github.com/rootless-containers/rootlesskit/pkg/common"
+	"github.com/rootless-containers/rootlesskit/pkg/network"
+)
+
+func NewParentDriver(binary string, mtu int, bridge string) (network.ParentDriver, error) {
+	if binary == "" {
+		return nil, errors.New("got empty binary")
+	}
+	if mtu < 0 {
+		return nil, errors.New("got negative mtu")
+	}
+	if mtu == 0 {
+		mtu = 1500
+	}
+	if bridge == "" {
+		return nil, errors.New("got empty bridge")
+	}
+	return &parentDriver{
+		binary: binary,
+		mtu:    mtu,
+		bridge: bridge,
+	}, nil
+}
+
+const opaqueDev = "lxcusernic.dev"
+
+type parentDriver struct {
+	binary string
+	mtu    int
+	bridge string
+}
+
+func (d *parentDriver) MTU() int {
+	return d.mtu
+}
+
+func (d *parentDriver) ConfigureNetwork(childPID int, stateDir string) (*common.NetworkMessage, func() error, error) {
+	var cleanups []func() error
+	dummyLXCPath := "/dev/null"
+	dummyLXCName := "dummy"
+	dev := "eth0"
+	cmd := exec.Command(d.binary, "create", dummyLXCPath, dummyLXCName, strconv.Itoa(childPID), "veth", d.bridge, dev)
```

Comment:
@brauner @stgraber

Is it valid to call `lxc-user-net create $lxcpath $name $pid $type $bridge $nicname` with `$lxcpath="/dev/null"` `$name="dummy"`?

IIUC, `lxc-user-net` just ignores `$lxcpath` and `$name`, right? https://github.com/lxc/lxc/blob/master/src/lxc/cmd/lxc_user_nic.c

Reply:
Yes, this should be valid. Let me add in a check that it is non-null so that we don't regress users in the future but calling it with a dummy path is fine.

Reply:
thanks!